src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -189,6 +282,63 @@ private static void RemoveUninstalledPackages(PackagesLockFile lockFile, IEnumer
             }
         }
 
+        public static PackagesLockFile FromPackagesConfigFile(
+            string pcFile,
+            NuGetFramework projectTfm,
+            string packagesFolderPath,
+            CancellationToken token)
+        {
+            if (pcFile == null)
+            {
+                throw new ArgumentNullException(nameof(pcFile));
+            }
+            if (!File.Exists(pcFile))
+            {
+                throw new FileNotFoundException("packages.config file does not exist", pcFile);
+            }
+            if (projectTfm == null)
+            {
+                throw new ArgumentNullException(nameof(projectTfm));
+            }
+            if (packagesFolderPath == null)
+            {
+                throw new ArgumentNullException(nameof(packagesFolderPath));
+            }
+            if (!Directory.Exists(packagesFolderPath))
+            {
+                throw new DirectoryNotFoundException("Packages directory does not exist");
+            }
+
+            var lockFile = new PackagesLockFile();
+            var target = new PackagesLockFileTarget();
+            lockFile.Targets.Add(target);
+            target.TargetFramework = projectTfm;
+
+            using (var stream = File.OpenRead(pcFile))
+            {
+                var contentHashUtil = new PackagesConfigContentHashProvider(new FolderNuGetProject(packagesFolderPath));
```

Comment:
If N projects have a package, wouldn't you hit the disk N times to read the sha? 

We don't optimize for that on the PR side as far as I am aware, but just wanted to bring it to your attention.

Reply:
yes, that's right, the `.nupkg.metadata` file will be read multiple times. I will be doing some basic performance testing before the feature branch gets merged to dev, so if it turns out to be a problem there will be another PR before it all gets merged to dev.